hercules-ci-api/src/Hercules/API/Build/FailureGraph.hs

```diff
@@ -0,0 +1,14 @@
+{-# LANGUAGE DeriveAnyClass #-}
+module Hercules.API.Build.FailureGraph where
+
+import           Hercules.API.Prelude
+import           Hercules.API.Derivation        ( Derivation )
+
+data Graph = Graph
+ { nodes :: [Node]
+ } deriving (Generic, Show, Eq, ToJSON, FromJSON, ToSchema)
+
+data Node = Node
+ { derivation :: Derivation
+ , failedDependencies :: [Text]
```

Comment:
What is the `[Text]` - store paths? Could we make these `Derivation`s? Or maybe just a comment if not :)